### PR TITLE
Add protobuf package

### PR DESCRIPTION
**Pull request by bartbroere**
Adds the Python `protobuf` package. I'll keep it marked as a draft until I'm sure it's okay and actually works as intended.

Having this package in place would open up a lot of possibilities for other packages. Backwards compatibility of protobufs seems okay, so I don't see any dangers for compatibility in having the newest release included in Pyodide. See related discussion in: https://github.com/pyodide/pyodide/issues/3424

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

**Comment by ryanking13** [comment]
Thanks for the porting work @bartbroere. I have some novice questions about protobuf:

1. What is the difference between, protobuf library / protobuf Python package / protoc binary? Does protobuf Python package compatible with other versions (e.g. protobuf <= 3.X) of protobuf / protoc?
2. What is the difference between protobuf pure Python package / C-extension package?

**Comment by bartbroere** [comment]
First of all, I still have some of the same questions, that's why I'll keep this pull request a draft for now.

> 1. What is the difference between, protobuf library / protobuf Python package / protoc binary? Does protobuf Python package compatible with other versions (e.g. protobuf <= 3.X) of protobuf / protoc?

The `protoc` binary originally is a C++ executable I believe, but there also is a Cython executable that imports `libprotoc` and is probably functionally equivalent, with the added bonus that it can be called from Python code directly without `subprocess.call` trickery. This last `protoc` flavour is in `grpcio_tools`, a package that I hope to add after this.

> 2. What is the difference between protobuf pure Python package / C-extension package?

I'm not sure, but I see adding the `protobuf` package as the first step towards adding more useful packages, such as `onnx`

**Comment by bartbroere** [comment]
`protobuf` isn't a pure Python wheel by the way, it seems, but it can be.

Like other packages it [has a fallback mechanism](https://github.com/protocolbuffers/protobuf/blob/main/python/google/protobuf/internal/api_implementation.py#L73-L79)
I added a test that checks whether the binary component is present.

And regarding `onnx`, this will be fixed in a release sometime soon I expect: https://github.com/onnx/onnx/pull/4956/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552

So, to summarise, I think it all looks okay to merge.

**Review by ryanking13 — APPROVED** [review]
Thanks, @bartbroere!
